### PR TITLE
The problem's points are now updated when the submission grid's refresh button is clicked.

### PR DESCRIPTION
**Pull request by deyordanov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1387

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx [hunk 1]
```diff
@@ -148,6 +149,29 @@ const ContestSolutionSubmitPage = () => {
...
+        if (submissionsData?.items && problems && submissionsData.items.length > 0) {
+            // eslint-disable-next-line max-len
+            const latestSubmission = submissionsData.items.reduce((newest, current) => new Date(current.createdOn) > new Date(newest.createdOn)
+                ? current
+                : newest, submissionsData.items[0]);
+
+            if (latestSubmission?.problem) {
+                const probIndex = problems.findIndex((p) => p.id === latestSubmission.problem.id);
+                if (probIndex !== -1) {
+                    const updatedProb = { ...problems[probIndex] };
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx:162]: We generally don't like abbreviations and encourage full names for variables -> `updatedProblem`

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx [hunk 2]
```diff
@@ -344,13 +368,13 @@ const ContestSolutionSubmitPage = () => {
...
+    const sumMyPoints = useMemo(() => updatedProblems
+        ? updatedProblems.reduce((accumulator, problem) => accumulator + problem.points, 0)
+        : 0, [ updatedProblems ]);
 
-    const sumAllContestPoints = useMemo(() => contest
-        ? contest.problems.reduce((accumulator, problem) => accumulator + problem.maximumPoints, 0)
-        : 0, [ contest ]);
+    const sumAllContestPoints = useMemo(() => updatedProblems
+        ? updatedProblems.reduce((accumulator, problem) => accumulator + problem.maximumPoints, 0)
+        : 0, [ updatedProblems ]);
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx:377]: What will happen if `updatedProblems` are not set, but there are problems and the user has points for them, will they be shown? Note: 0 submissions with points for the contest is a valid case (if submissions are deleted).

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx (hunk 1 above)
Comment (deyordanov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx (hunk 2 above)
Comment (deyordanov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx]: Done.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**